music_assistant/providers/audioscrobbler/__init__.py

```diff
@@ -0,0 +1,315 @@
+            CONF_SESSION_KEY: self.config.get_value(CONF_SESSION_KEY),
+        }
+
+    async def loaded_in_mass(self) -> None:
+        """Call after the provider has been loaded."""
+        await super().loaded_in_mass()
+
+        if not self.config.get_value(CONF_SESSION_KEY):
+            self.logger.info("No session key available")
+            return
+
+        self._network = _get_network(self._get_network_config())
+
+        # subscribe to internal events
+        self.mass.subscribe(self._on_mass_queue_updated, EventType.QUEUE_UPDATED)
+        self.mass.subscribe(self._on_mass_media_item_played, EventType.MEDIA_ITEM_PLAYED)
+        self.logger.debug("subscribed to events")
+
+    async def _on_mass_queue_updated(self, event: MassEvent) -> None:
+        """Player has updated, update nowPlaying."""
+        if self._network is None:
+            self.logger.error("no network available during _on_mass_queue_updated")
+            return
+
+        if self._processing_queue_update:
+            return
+
+        queue: PlayerQueue = event.data
+        if queue.state != PlayerState.PLAYING or not is_track(queue.current_item.media_item):
+            self.logger.debug("queue update ignored, no track currently playing")
+            self._currently_playing = None
+            return
+
+        track: Track = queue.current_item.media_item
+        if track.uri == self._currently_playing:
+            self.logger.debug(
+                f"queue update ignored, track {track.uri} already marked as 'now playing'"
+            )
+            return
+
+        self._processing_queue_update = True
+
+        def update_now_playing() -> None:
+            try:
+                self._network.update_now_playing(
+                    track.artist_str,
+                    track.name,
+                    track.album.name if track.album else None,
+                    track.album.artist_str if isinstance(track.album, Album) else None,
+                    track.duration,
+                    track.track_number,
+                    track.mbid,
+                )
+                self.logger.debug(f"track {track.uri} marked as 'now playing'")
+                self._currently_playing = track.uri
+            except Exception as err:
+                self.logger.exception(err)
+            finally:
+                self._processing_queue_update = False
+
+        self.mass.loop.run_in_executor(None, update_now_playing)
+
+    async def _on_mass_media_item_played(self, event: MassEvent) -> None:
+        """Media item has finished playing, we'll scrobble the track."""
+        if self._network is None:
+            self.logger.error("no network available during _on_mass_media_item_played")
+            return
+
+        item = await self.mass.music.get_item_by_uri(event.data["media_item"])
+        if not is_track(item):
+            self.logger.debug(f"track {event.data['media_item']} skipped, as it is not a track")
+            return
```

Comment:
I'm not following why its needed to request a full item details (which is also a potential call to a streaming provider api) - lets try to prevent this and work with the event data we already have